rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/common/SchedulerRestInterface.java

```diff
@@ -456,6 +539,75 @@ TaskStateDataPage getJobTaskStatesByTagPaginated(
+     * 
+     * @param sessionId  a valid session id.
+     * @param from  the scheduled date to which we start fetching tasks. The format is in Epoch time.
+     * @param to  the end scheduled end date to stop fetching tasks. The format is in Epoch time.
+     * @param mytasks  <code>True</code> if you want to fetch only the user's tasks. Default value is <code>False</code>.
+     * @param running  fetch running tasks. Default value is <code>True</code>.
+     * @param pending  fetch pending tasks. Default value is <code>True</code>.
+     * @param finished  fetch finished tasks. Default value is <code>True</code>.
+     * @param offset  the index of the first task to fetch (for pagination).
+     * @param limit  the index of the last (excluded) task to fetch (for pagination).
+     * @return a list of <code>TaskStateData</code>  and the total number of them.
+     */
+    @GET
+    @GZIP
+    @Path("taskstates")
+    @Produces("application/json")
+    RestPage<TaskStateData> getTaskStates (
+            @HeaderParam("sessionid") String sessionId,
+            @QueryParam("from") @DefaultValue("0") long from,
+            @QueryParam("to") @DefaultValue("0") long to,
+            @QueryParam("mytasks") @DefaultValue("false") boolean mytasks,
+            @QueryParam("running") @DefaultValue("true") boolean running,
+            @QueryParam("pending") @DefaultValue("true") boolean pending,
+            @QueryParam("finished") @DefaultValue("true") boolean finished,
+            @QueryParam("offset") @DefaultValue("0") int offset,
+            @QueryParam("limit") @DefaultValue("-1") int limit)
+                    throws NotConnectedRestException, PermissionRestException;
+    
+    /**
+     * Returns a paginated list of <code>TaskStateData</code> regarding the given parameters (decoupled from the associated jobs).
+     * The result is paginated using the optional <code>offset</code> and <code>limit</code> parameters.
+     * If those parameters are not specified, the following values will be used: [0, DEFAULT_VALUE[
+     * The DEFAULT_VALUE can be set in the scheduler config file as the <code>pa.scheduler.tasks.page.size</code> parameter.
+     * 
+     * @param sessionId  a valid session id.
+     * @param taskTag  tag to filter the tasks. The tag should be complete as the criteria is strict.
+     * @param from  the scheduled date to which we start fetching tasks. The format is in Epoch time.
+     * @param to  the end scheduled end date to stop fetching tasks. The format is in Epoch time.
+     * @param mytasks <code>True</code> if you want to fetch only the user's tasks. <code>False</code> will fetch everything.
+     * @param running  fetch running tasks. Default value is <code>True</code>.
+     * @param pending  fetch pending tasks. Default value is <code>True</code>.
+     * @param finished  fetch finished tasks. Default value is <code>True</code>.
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)